### ISSUE TITLE
Imagenet: Allow to download validation set only.

### ISSUE THREAD

**Issue by MiWeiss** [tensorflow/datasets#2286]
**Is your feature request related to a problem? Please describe.**
To use the validation set of [Imagenet2012](https://www.tensorflow.org/datasets/catalog/imagenet2012) by calling `tfds.load('imagenet2012', split='validation')`, tfds requires that both training and validation set are downloaded. 

Note that the training set is quite large and imagenet downloads are quite slow. Thus downloading and storing if not used is undesirable.

**Describe the solution you'd like**
If only one of the two datasets (either training or validation) is found on the file system, tfds prints a warning. 
Only if access to one which is not stored is attempted, e.g. by calling `tfds.load('imagenet2012', split='train+validation')` an error should be raised

**Comment by NikhilBartwal** [comment]
Hey @MiWeiss, 
That is, indeed, an efficient solution to replace the AssertionError with two different ones, depending on the split user selected otherwise, it would just waste user's time.
Could you tell about the status of this feature?

**Comment by NikhilBartwal** [comment]
Hey @MiWeiss,
That would, indeed, be a good optimization but currently the way TFDS loads a dataset is by first going through all of the splits, downloading and preparing the necessary tfrecord files and then provide by split, as requested by the user.
This is done so as in the future, if the user needs a different split, he could easily do that without TFDS having to download that very split at runtime.
If we want to enhance it such that TFDS downloads and prepares every unique split only when it is requested by the user, it will need some major changes in the way `tfds.core.dataset_builder.py` and `tfds.core.load.py` works.
@Conchylicultor What do you think about this?

**Comment by ibarrond** [comment]
Same Issue. I tried creating a "fake" training dataset, but apparently there is some checksum being aplied to make sure that the file is correct. Maybe an empty `.tar` file with a hardcoded checksum would do the trick?

**Comment by Conchylicultor** [comment]
We could add a condition, similar to the one to skip the test set if not present: https://github.com/tensorflow/datasets/blob/c7439f84c2a77891e07a8e154cbde21f6ddcc8e9/tensorflow_datasets/image_classification/imagenet.py#L211

Don't hesitate to send a PR.

**Comment by ibarrond** [comment]
PR sent!

**Comment by MiWeiss** [comment]
Looks like a good solution, allowing us to close this now. Thanks for PR & Review @ibarrond @Conchylicultor .

Feel free to re-open issue if I missed something and #2484 did not completely solve this issue.
